java/lang/Float/FloatTest.cpp

```diff
@@ -24,538 +24,1578 @@
+    ASSERT_EQUAL(-1, Float::compare(Float::MIN_VALUE, Float::NaN));
+    ASSERT_EQUAL(-1, Float::compare(Float::MIN_VALUE, Float::POSITIVE_INFINITY));
+//    ASSERT_EQUAL(Float::compare(Float::MIN_VALUE, Float::MAX_VALUE));
+    ASSERT_EQUAL(-1, Float::compare(Float::MIN_VALUE, 1.2));
+    ASSERT_EQUAL(1, Float::compare(Float::MIN_VALUE, 0.0));
+    ASSERT_EQUAL(1, Float::compare(Float::MIN_VALUE, -0.0f));
+    ASSERT_EQUAL(1, Float::compare(Float::MIN_VALUE, -1.2f));
+    ASSERT_EQUAL(0, Float::compare(Float::MIN_VALUE, Float::MIN_VALUE));
+    ASSERT_EQUAL(1, Float::compare(Float::MIN_VALUE, Float::NEGATIVE_INFINITY));
+
+    // Test value 1.2
+    ASSERT_EQUAL(-1, Float::compare(1.2, Float::NaN));
+    ASSERT_EQUAL(-1, Float::compare(1.2, Float::POSITIVE_INFINITY));
+//    ASSERT_EQUAL(-1, Float::compare(1.2, Float::MAX_VALUE));
+    ASSERT_EQUAL(0, Float::compare(1.2, 1.2));
+    ASSERT_EQUAL(1, Float::compare(1.2, 0.0));
+    ASSERT_EQUAL(1, Float::compare(1.2, -0.0f));
+    ASSERT_EQUAL(1, Float::compare(1.2, -1.2f));
+    ASSERT_EQUAL(1, Float::compare(1.2, Float::MIN_VALUE));
+    ASSERT_EQUAL(1, Float::compare(1.2, Float::NEGATIVE_INFINITY));
+
+    // Test value 0.0
+    ASSERT_EQUAL(-1, Float::compare(0.0, Float::NaN));
+    ASSERT_EQUAL(-1, Float::compare(0.0, Float::POSITIVE_INFINITY));
+//    ASSERT_EQUAL(Float::compare(0.0, Float::MAX_VALUE));
+    ASSERT_EQUAL(-1, Float::compare(0.0, 1.2));
+    ASSERT_EQUAL(0, Float::compare(0.0, 0.0));
+    ASSERT_EQUAL(0, Float::compare(0.0, -0.0f));  // error in OSX
+    ASSERT_EQUAL(1, Float::compare(0.0, -1.2f));
+    ASSERT_EQUAL(-1, Float::compare(0.0, Float::MIN_VALUE));
+    ASSERT_EQUAL(1, Float::compare(0.0, Float::NEGATIVE_INFINITY));
+
+    // Test value -0.0f
+    ASSERT_EQUAL(-1, Float::compare(-0.0f, Float::NaN));
+    ASSERT_EQUAL(-1, Float::compare(-0.0f, Float::POSITIVE_INFINITY));
+//    ASSERT_EQUAL(Float::compare(-0.0f, Float::MAX_VALUE));
+    ASSERT_EQUAL(-1, Float::compare(-0.0f, 1.2));
+    ASSERT_EQUAL(0, Float::compare(-0.0f, 0.0));  // error in OSX
+    ASSERT_EQUAL(0, Float::compare(-0.0f, -0.0f));   // error in OSX
+    ASSERT_EQUAL(1, Float::compare(-0.0f, -1.2f));
+    ASSERT_EQUAL(-1, Float::compare(-0.0f, Float::MIN_VALUE));
+    ASSERT_EQUAL(1, Float::compare(-0.0f, Float::NEGATIVE_INFINITY));
+
+    // Test value -1.2f
+    ASSERT_EQUAL(-1, Float::compare(-1.2f, Float::NaN));
+    ASSERT_EQUAL(-1, Float::compare(-1.2f, Float::POSITIVE_INFINITY));
+//    ASSERT_EQUAL(Float::compare(-1.2f, Float::MAX_VALUE));
+    ASSERT_EQUAL(-1, Float::compare(-1.2f, 1.2));
+    ASSERT_EQUAL(-1, Float::compare(-1.2f, 0.0));
+    ASSERT_EQUAL(-1, Float::compare(-1.2f, -0.0f));
+    ASSERT_EQUAL(0, Float::compare(-1.2f, -1.2f));
+    ASSERT_EQUAL(-1, Float::compare(-1.2f, Float::MIN_VALUE));
+    ASSERT_EQUAL(1, Float::compare(-1.2f, Float::NEGATIVE_INFINITY));
+}
+
+// TODO(thoangminh): Need to pass all test cases in comment block
+TEST (JavaLang, FloatEquals) {
```

Comment:
Please remove some duplicate test

Reply:
Please explain more which test need to remove.